.github/container/Dockerfile.base

```diff
@@ -29,6 +29,18 @@ FROM ${BASE_IMAGE}
 ARG GIT_USER_EMAIL
 ARG GIT_USER_NAME
 ARG CLANG_VERSION
+ARG JAX_NCCL_VERSION
```

Comment:
Where would a default be specified? I think the situation today is that we want the default nightly/CI behaviour to be "12.6.1 base image with NCCL 2.23 installed on top", so it should be possible to express that.

I know that for the base image, there is special treatment of `latest` to enable this.

Reply:
So the "special treatment" is to embed a default  in the Dockerfile, and then conditionally override it via an Actions conditional expression:
https://github.com/NVIDIA/JAX-Toolbox/blob/61d8446ce734799538c5124db7631ccf517f4bc1/.github/workflows/_build_base.yaml#L135